zulipterminal/helper.py

```diff
@@ -399,8 +404,44 @@ def index_messages(messages: List[Message], model: Any, index: Index) -> Index:
     narrow = model.narrow
     for msg in messages:
         if "edit_history" in msg:
-            index["edited_messages"].add(msg["id"])
-
+            for edit_history_event in msg["edit_history"]:
+                if "prev_content" in edit_history_event:
+                    index["edited_messages"].add(msg["id"])
+                if "prev_stream" in edit_history_event:
+                    index["moved_messages"].add(msg["id"])
+                if "prev_topic" in edit_history_event:
+                    # We know it has a topic edit. Now we need to determine if
+                    # it was a true move or a resolve/unresolve.
+                    if not edit_history_event["topic"].startswith(
+                        RESOLVED_TOPIC_PREFIX
+                    ):
+                        if (
+                            edit_history_event["prev_topic"].startswith(
+                                RESOLVED_TOPIC_PREFIX
+                            )
+                            and edit_history_event["prev_topic"][2:]
+                            != edit_history_event["topic"]
+                        ):
+                            index["moved_messages"].add(msg["id"])
+                        if not edit_history_event["prev_topic"].startswith(
+                            RESOLVED_TOPIC_PREFIX
+                        ) and not edit_history_event["topic"].startswith(
+                            RESOLVED_TOPIC_PREFIX
+                        ):
+                            index["moved_messages"].add(msg["id"])
+                    else:
+                        if (
+                            edit_history_event["prev_topic"].startswith(
+                                RESOLVED_TOPIC_PREFIX
+                            )
+                            and edit_history_event["prev_topic"][2:]
+                            != edit_history_event["topic"][2:]
```

Comment:
I think you have not handled moved message case if message is edited to resolve/unresolve topic with same name.